lib/queryBuilder/RelationExpression.js

```diff
@@ -185,9 +186,15 @@ class RelationExpression {
       for (let i = 0, l = childNames.length; i < l; ++i) {
         const childName = childNames[i];
         const node = this[childName];
-        const relation = allRelations[node.$relation] || null;
-        const childExpr = new RelationExpression(node, 0);
+        const relation = allRelations[node.$relation];
+        if (!relation) {
+          throw new ValidationError({
```

Comment:
I don't think this class should throw a `ValidationError` directly, but some other error and the catched can decide which error is thrown out from objection. Let's keep this class as low-level as possible.

Reply:
> the catched can decide which error is thrown out from objection

Could you please clarify this part? Should all places that call this code have catch block for rethrowing specific errors?

Reply:
It should say "catcher" instead of "catched". I just mean that we should create some custom error and throw that and then the code that catches the error should convert it to a `ValidationError`. This doesn't work if the user has overwritten the `Model.createValidationError` method.

Reply:
It's called from two places and the other one doesn't have any error handling currently. You can just leave it as is. So really, it's just a matter of moving that code to a new place.

Reply:
Pushed changes, hopefully I understood your idea correctly :)